soc/realtek/ec/rts5912/soc.c

```diff
@@ -28,11 +28,13 @@ bool z_arm_on_enter_cpu_idle(void)
  */
 void soc_early_init_hook(void)
 {
+#ifdef CONFIG_RTS5912_DEBUG_SWJ
```

Comment:
It is recommended to use the IS_ENABLED macro whenever possible.  This allows static analysis tools to do more checks.

Please try out the version shown below and confirm that the build and link works when the SWJ node is disabled.

```
void soc_early_init_hook(void)
{
	if (!IS_ENABLED(CONFIG_RTS5912_DEBUG_SWJ)) {
		return;
	}

	int ret;

	/* Apply device related preinit configuration */
	ret = swj_connector_init();
	if (ret < 0) {
		LOG_ERR("SWJ init failed");
	}
}
```